catalog/spatial/registry/registry-federation-admin-service-impl/src/test/java/org/codice/ddf/registry/federationadmin/service/impl/IdentityNodeInitializationTest.java

```diff
@@ -0,0 +1,157 @@
+import java.io.InputStream;
+import java.util.Optional;
+
+import org.codice.ddf.configuration.SystemInfo;
+import org.codice.ddf.parser.ParserException;
+import org.codice.ddf.parser.xml.XmlParser;
+import org.codice.ddf.registry.common.metacard.RegistryObjectMetacardType;
+import org.codice.ddf.registry.federationadmin.service.FederationAdminException;
+import org.codice.ddf.registry.schemabindings.helper.MetacardMarshaller;
+import org.codice.ddf.registry.transformer.RegistryTransformer;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.transform.CatalogTransformerException;
+import ddf.security.Subject;
+import oasis.names.tc.ebxml_regrep.xsd.rim._3.RegistryPackageType;
+
+@RunWith(MockitoJUnitRunner.class)
+public class IdentityNodeInitializationTest {
+
+    private static final String TEST_SITE_NAME = "Slate Rock and Gravel Company";
+
+    private static final String TEST_VERSION = "FF 2.0";
+
+    @Mock
+    private FederationAdminServiceImpl federationAdminService;
+
+    private IdentityNodeInitialization identityNodeInitialization;
+
+    private Metacard testMetacard;
+
+    private RegistryTransformer registryTransformer;
+
+    @Mock
+    private Subject subject;
+
+    private XmlParser parser;
+
+    private MetacardMarshaller metacardMarshaller;
+
+    @Before
+    public void setUp() throws Exception {
+        parser = spy(new XmlParser());
+        identityNodeInitialization = spy(new IdentityNodeInitialization());
+        registryTransformer = spy(new RegistryTransformer());
+        metacardMarshaller = spy(new MetacardMarshaller(parser));
+        registryTransformer.setParser(parser);
+        identityNodeInitialization.setRegistryTransformer(registryTransformer);
+        identityNodeInitialization.setMetacardMarshaller(metacardMarshaller);
+        identityNodeInitialization.setFederationAdminService(federationAdminService);
+        System.setProperty(SystemInfo.SITE_NAME, TEST_SITE_NAME);
+        System.setProperty(SystemInfo.VERSION, TEST_VERSION);
+        testMetacard = getTestMetacard();
+    }
+
+    @Test
+    public void initWithNoPreviousEntry() throws Exception {
+        ArgumentCaptor<Metacard> captor = ArgumentCaptor.forClass(Metacard.class);
+        when(federationAdminService.getLocalRegistryIdentityMetacard()).thenReturn(Optional.empty());
+        identityNodeInitialization.init();
+        verify(federationAdminService).addRegistryEntry(captor.capture());
+        Metacard metacard = captor.getValue();
+        assertThat(metacard.getAttribute(Metacard.MODIFIED), notNullValue());
+        assertThat(metacard.getAttribute(Metacard.CREATED), notNullValue());
+        assertThat(metacard.getAttribute(RegistryObjectMetacardType.REGISTRY_IDENTITY_NODE),
```

Comment:
For modified and create, the values are dependent on when the system it runs since it just takes the current time. I can try a test that checks if the time the attributes have been created is after a time I set myself. 

I changed the bottom two assertThat to equal true.